firmware/Core/Src/unit_tests/unit_test_helpers.c

```diff
@@ -1,2 +1,57 @@
 #include "unit_tests/unit_test_helpers.h"
+#include "telecommands/telecommand_args_helpers.h"
+#include "string.h"
 
+uint8_t TEST_EXEC__TCMD_hex_string_to_byte_array(){
+    char normal[9] = "FFFFFFFF\0";
+    char normal_delim[13] = "FF_FFAA 01\0";
+    char normal_args_and_uneven[22] = "FF,012200FF,FFF\0";
+    char not_even[3]="FFF";
+    char not_even_delimiter[5]="FF_F\0";
+    char header_args[10] = "0xFF,0xFF\0";
+    // uint8_t expected_answer[2][4] = {
+    //     {255,255,255,255},
+    //     {1,22,0,255}
+    // };
+    // uint8_t answer[4] = {255,255,255,255};
+    uint8_t result[4];
+    // uint8_t result[3];
+    uint8_t val = 0;
+
+    TEST_ASSERT(TCMD_extract_hex_array_arg(normal,0,result,sizeof(result))==0); // tests if function has no error
+
+    for(int i=0;i<4;i++){
+        if (result[i]!=255){
+            val = 1;
+        }
+    }
+
+    TEST_ASSERT(TCMD_extract_hex_array_arg(normal_delim,0,result,sizeof(result))==0); // tests if function has no error
+
+    if (result[0]!=255){val=1;} 
```

Comment:
A better way to write this is `TEST_ASSERT(result[0] == 0xFF);`, `TEST_ASSERT(result[1] == 0xFF);`, etc.